extensions/sql-database-projects/src/common/utils.ts

```diff
@@ -3,7 +3,40 @@
  *  Licensed under the Source EULA. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
+import * as vscode from 'vscode';
 
 export function getErrorMessage(error: Error | string): string {
 	return (error instanceof Error) ? error.message : error;
 }
+
+export function trimUri(innerUri: vscode.Uri, outerUri: vscode.Uri): string {
+	let innerParts = innerUri.path.split('/');
+	let outerParts = outerUri.path.split('/');
+
+	while (innerParts.length > 0 && outerParts.length > 0 && innerParts[0].toLocaleLowerCase() === outerParts[0].toLocaleLowerCase()) {
+		innerParts = innerParts.slice(1);
+		outerParts = outerParts.slice(1);
+	}
+
+	return outerParts.join('/');
+}
+
+declare global {
+	interface String {
+		trimChars(chars: string): string;
+	}
+}
+
+String.prototype.trimChars = function (chars: string): string {
```

Comment:
You shouldn't modify global prototypes.  Instead just make a helper function that does the same thing but takes the string as an input.

Reply:
+1

---
In reply to: [369281017](https://github.com/microsoft/azuredatastudio/pull/8921#discussion_r369281017) [](ancestors = 369281017)